client/core/core.go

```diff
@@ -1373,79 +1374,100 @@ func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) []*DEXBrief {
-				result.AuthErr = details
-				continue
+	var enabledDEXConnections = make(map[string]*dexConnection)
+	for host, dc := range c.conns {
+		wg.Add(1)
+		go func(host string, dc *dexConnection) {
+			defer wg.Done()
+			dc.tradeMtx.RLock()
+			tradeIDs := make([]string, 0, len(dc.trades))
+			for tradeID := range dc.trades {
+				tradeIDs = append(tradeIDs, tradeID.String())
 			}
-			// Try to unlock the Decred wallet, which should run the reFee cycle, and
-			// in turn will run authDEX.
-			dcrWallet, err := c.connectedWallet(dcrID)
-			if err != nil {
-				log.Debugf("Failed to connect for reFee at %s with error: %v", dc.acct.host, err)
-				details := fmt.Sprintf("Incomplete registration detected for %s, but failed to connect to the Decred wallet", dc.acct.host)
-				c.notify(newFeePaymentNote("Wallet connection warning", details, db.WarningLevel, dc.acct.host))
-				result.AuthErr = details
-				continue
+			dc.tradeMtx.RUnlock()
+			result := &DEXBrief{
+				AcctID:   dc.acct.ID().String(),
+				Host:     dc.acct.host,
+				TradeIDs: tradeIDs,
+				Authed:   false,
 			}
-			if !dcrWallet.unlocked() {
-				err = unlockWallet(dcrWallet, crypter)
-				if err != nil {
-					details := fmt.Sprintf("Connected to Decred wallet to complete registration at %s, but failed to unlock: %v", dc.acct.host, err)
-					c.notify(newFeePaymentNote("Wallet unlock error", details, db.ErrorLevel, dc.acct.host))
-					result.AuthErr = details
-					continue
-				}
+			results = append(results, result)
+			err := dc.acct.unlock(crypter)
+			if err != nil {
+				details := fmt.Sprintf("error unlocking account for %s: %v", dc.acct.host, err)
+				c.notify(newFeePaymentNote("Account unlock error", details, db.ErrorLevel, dc.acct.host))
+				return
 			}
-			c.reFee(dcrWallet, dc)
-			continue
-		}
-
-		wg.Add(1)
-		go func(dc *dexConnection) {
-			defer wg.Done()
-			err := c.authDEX(dc)
+			acctInfo, err := c.db.Account(dc.acct.id)
 			if err != nil {
-				details := fmt.Sprintf("%s: %v", dc.acct.host, err)
+				log.Errorf("Error retrieving account: %v", err)
+				return
+			}
+			if acctInfo.Disabled {
+				return
+			}
+			authDEXErr := c.authDEX(dc)
+			var mErr *msgjson.Error
+			// Disable account on all authentication errors, including when the server account is
+			// marked as unpaid while the client account is marked as paid
+			if errors.As(authDEXErr, &mErr) &&
+				(mErr.Code != msgjson.UnpaidAccountError ||
+					(mErr.Code == msgjson.UnpaidAccountError && dc.acct.feePaid())) {
+				details := fmt.Sprintf("%s: %v", dc.acct.host, authDEXErr)
 				c.notify(newFeePaymentNote("DEX auth error", details, db.ErrorLevel, dc.acct.host))
 				result.AuthErr = details
+				err = c.db.DisableAccount(acctInfo)
+				if err != nil {
+					log.Errorf("Error disabling account: %v", err)
+				}
 				return
 			}
 			result.Authed = true
-		}(dc)
+			enabledDEXConnections[host] = dc
```

Comment:
Yes, that looks like a problem. I thought it would be safe since the same host would never be used per parallel process but it looks like that is indeed a no no.